File: BOJ/13705.cpp
#include <bits/stdc++.h>
using namespace std;
#define int long long 
#define double long double

double A,B,C;

double Tau=6.2831853071795864768;

double sin_(double x){
    x+=Tau;
    x=x-Tau*(int)(x/Tau);
    double ret=0;
    for(int i=0;i<500;i++){
        double now=(i%2==0?1:-1);
        for(double j=1;j<=i*2+1;j++){
            now*=x;
            now/=j;
        }
        ret+=now;
    }
    return ret;
}
double f(double x){
    //cout<<sin(x)<<' '<<sin_(x)<<endl;
    return A*x+B*sin_(x)-C;
}

int32_t main(){
    ios_base::sync_with_stdio(false);
    cin.tie(0);
    cout.tie(0);
    
    cout<<fixed;
    cout.precision(6);

    cin>>A>>B>>C;
    
    //f(x)=Ax+Bsin(x)-C

    double s=0,e=200000,m;
    while(e-s>1e-9){
        m=(s+e)/2;
        double now=f(m);
        //cout<<m<<"-------"<<now<<endl;
        if(now>0) e=m+1e-10;
        else if(now<0) s=m-1e-10;
        else break;
    }
    cout<<m<<endl;

}

File: BOJ/13705.py
from decimal import Decimal

a,b,c=input().split()
a=Decimal(a)
b=Decimal(b)
c=Decimal(c)


s=Decimal('-1')
e=Decimal('200000')
m=Decimal('0')

Tau=Decimal('6.283185307179586476925286766559005768394338798750211641949889184615632812572417997256069650684234135964296173026564613294187689219101164463450718816256')


def sin(x):
    x=x+Tau-Tau*int((x+Tau)/Tau)
    ret=Decimal('0')
    now=Decimal(x)
    for i in range(1000):
        ret+=now
        now*=-x*x
        now/=(2*i+3)*(2*i+2)    
    return ret
    
def f(x):
    return Decimal(a*x+b*sin(x)-c)    

while(e-s>=Decimal('1e-20')):
    m=Decimal((s+e)/2)
    #print(float(e-s))
    now=f(m)
    if now>0:
        e=m
    elif now<0:
        s=m
    else:
        break 
    
print(round(m,6))